doc/tools/docgram/orderedGrammar

```diff
@@ -1157,6 +1157,7 @@ section_var_expr50: [
 
 section_var_expr0: [
 | starred_ident_ref
+| "()"
```

Comment:
I expect you added this because it's necessary for your fix.  But since there's no reason for users to ever use "()", how about dropping this production in `common.edit_mlg` so it's not shown in the doc?  Or, failing that, move it down one line here as its less important than `| "(" section_var_expr ")" OPT "*"`.

FWIW, "(  )" would not be be valid.  The production would have to use "(" ")" to make that valid.

Reply:
>FWIW, "( )" would not be be valid. The production would have to use "(" ")" to make that valid.

I don't understand

Reply:
The grammar accepts the token "()".  It won't parse if there is white space in the middle.  Most people are accustomed to think of "(" and ")" as separate tokens.

Reply:
It does parse through `"(" section_var_expr ")" OPT "*"` because section_var_expr has a LIST0 case

Reply:
`with` is a keyword not an id

Reply:
True but not relevant.  This example has an empty section_subset_expr plus a `with` clause.

![image](https://user-images.githubusercontent.com/1253341/174162766-83f1e14a-2c5d-4e8c-8513-f7cd18d2bb26.png)

Reply:
The lookahead strikes again I guess

Reply:
I think it fails for any nonterminal that is empty IIRC.